jgrapht-core/src/main/java/org/jgrapht/alg/cycle/HierholzerEulerianCycle.java

```diff
@@ -235,21 +235,32 @@ private void initialize(Graph<V, E> g)
     /**
      * Release any memory held.
      */
-    private void cleanup()
+    protected void cleanup()
     {
         this.g = null;
         this.verticesHead = null;
         this.eulerianHead = null;
         this.startVertex = null;
     }
 
+    /**
+     * Get available edge for given vertex.
+     *
+     * @param currentVertex the vertex
+     * @param previousEdge the previous edge (can be null)
+     * @return available edge for vertex
+     */
+    protected EdgeNode getNextEdge(VertexNode currentVertex, EdgeNode previousEdge) {
```

Comment:
This method is not needed at all. If you want to adjust the behavior of `computePartialEdge()` you can do so by overriding the method in the subclass.